### PR TITLE
Update 1-sided permutation docstring & args name

### PR DESCRIPTION
**Pull request by FarnazH**
- Update the docstring to have the Procrustes formulas at the top
- Change the argument names `remove_zero_col` and `remove_zero_row` to `unpad_col` and `unpad_row`

### CODE REVIEW

File: procrustes/permutation.py [hunk 1]
```diff
@@ -77,57 +96,42 @@ def permutation(
...
+        new_a, new_b = _zero_padding(new_a, new_b, "square")
+
     # compute permutation Procrustes matrix
     array_p = np.dot(new_a.T, new_b)
-    array_c = np.full(array_p.shape, np.max(array_p))
-    array_c -= array_p
+    array_c = np.full(array_p.shape, np.max(array_p)) - array_p
     array_u = np.zeros(array_p.shape)
     # set elements to 1 according to Hungarian algorithm (linear_sum_assignment)
     array_u[linear_sum_assignment(array_c)] = 1
```
Comment (FarnazH) [inline procrustes/permutation.py:137]: Reading the docstring and comments (as well as the paper), it was not clear to me why the cost function is defined this way, so I think we need to add some clarification. @fwmeng88  can you please share the reference/notes used? Thanks.

Comment (fwmeng88) [reply procrustes/permutation.py:137]: The Hungarian assignment seeks for minimizing the assignment cost, `x=A<sup>T</sup>B`. But in our scenario, we want to maximize the cost, which is equivalently to maximize the overlap (permutation Procrustes). Therefore, we can say we want to minimize `-x`. But in some cases, there can be a negative value, therefore, we add all the elements in the cost function with the maximum value to ensure all the elements are positive. 

I am not sure if I explain it clear or not. Maybe @PaulWAyers have some ideas.
@FarnazH

Comment (PaulWAyers) [reply procrustes/permutation.py:137]: This is a lot clearer. Can we be sure the documentation explains this?

@fwmeng88 was this something you cooked up, or is there a reference? The reference I have doesn't mention the negative and shift, though it is kinda obvious we should still give credit if we took it from somewhere.

Comment (fwmeng88) [reply procrustes/permutation.py:137]: I didn't remember we referred some references for this problem settings. But when I checked up this morning, there is an old paper that mentioned this and we should cite this, [Variants of the Hungarian method for assignment problems](https://onlinelibrary.wiley.com/doi/abs/10.1002/nav.3800030404). The manuscript has been updated accordingly.

**Review by fwmeng88 — APPROVED** [review]

File: procrustes/permutation.py (hunk 1 above)
Comment (fwmeng88) [reply procrustes/permutation.py:137]: This code snippet shows a draft docstring that can be added for single-sided permutation Procrustes.

```reStructuredText
    This is a matching problem and can be solved by the Hungarian method. The Hungarian assignment
    algorithm seeks to minimize the assignment cost,
    :math:`\mathbf{C} = \mathbf{A}^T \mathbf{B}`. But a maximum permutation sum is
    required for permutation Procrustes where we used `-\mathbf{C} = \mathbf{A}^T \mathbf{B}` as our
    objective function. A second strategy we used here is to ensure the non-negativeness by adding
    the maximum element to the objective matrix.
```

I don't think it a wise choice to push commits to your branch, which can help keep the history clean. Hope it helps a little.
@FarnazH

Comment (FarnazH) [reply procrustes/permutation.py]: Thanks @fwmeng88. I used C=A.T B as the cost function, as the `scipy.optimize.linear_sum_assignment` had an argument for maximization, so I set `maximize=True`.

**Pull request merged**